### ISSUE TITLE
[sim-services] I3MCTree throws when printing after removing nodes (Trac #1044)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1038]
This is either in sim-services/I3InIceCORSIKATrimmer, dataclasses/I3MCTree, or perhaps (but not likely), in the test.

I'm sure it's related to erasing nodes in the tree, but a simply printing the tree gives the following error message :

python: /home/olivas/icecube/pandoras-icebox/trunk/src/dataclasses/public/dataclasses/physics/I3MCTree.h:224: ValueType& TreeBase::Tree<T, Key, Hash>::iterator_storage_impl<boost::optional<Key>, ValueType, TreeType, TreeHashMapIter>::dereference() const [with ValueType = const I3Particle; TreeType = const TreeBase::Tree<I3Particle, I3ParticleID>; TreeHashMapIter = __gnu_cxx::_Hashtable_const_iterator<std::pair<const I3ParticleID, TreeBase::TreeNode<I3Particle> >, I3ParticleID, __gnu_cxx::hash<I3ParticleID>, std::_Select1st<std::pair<const I3ParticleID, TreeBase::TreeNode<I3Particle> > >, std::equal_to<I3ParticleID>, std::allocator<TreeBase::TreeNode<I3Particle> > >; T = I3Particle; Key = I3ParticleID; Hash = __gnu_cxx::hash<I3ParticleID>]: Assertion `iter != treePtr_->internalMap.end()' failed.

I'll try to isolate the problem more tomorrow, but have to run now.

Migrated from https://code.icecube.wisc.edu/ticket/1044
```json
{
    "status": "closed", 
    "changetime": "2015-08-10T20:31:17", 
    "description": "This is either in sim-services/I3InIceCORSIKATrimmer, dataclasses/I3MCTree, or perhaps (but not likely), in the test.\n\nI'm sure it's related to erasing nodes in the tree, but a simply printing the tree gives the following error message :\n\npython: /home/olivas/icecube/pandoras-icebox/trunk/src/dataclasses/public/dataclasses/physics/I3MCTree.h:224: ValueType& TreeBase::Tree<T, Key, Hash>::iterator_storage_impl<boost::optional<Key>, ValueType, TreeType, TreeHashMapIter>::dereference() const [with ValueType = const I3Particle; TreeType = const TreeBase::Tree<I3Particle, I3ParticleID>; TreeHashMapIter = __gnu_cxx::_Hashtable_const_iterator<std::pair<const I3ParticleID, TreeBase::TreeNode<I3Particle> >, I3ParticleID, __gnu_cxx::hash<I3ParticleID>, std::_Select1st<std::pair<const I3ParticleID, TreeBase::TreeNode<I3Particle> > >, std::equal_to<I3ParticleID>, std::allocator<TreeBase::TreeNode<I3Particle> > >; T = I3Particle; Key = I3ParticleID; Hash = __gnu_cxx::hash<I3ParticleID>]: Assertion `iter != treePtr_->internalMap.end()' failed.\n\nI'll try to isolate the problem more tomorrow, but have to run now.\n", 
    "reporter": "olivas", 
    "cc": "david.schultz, jvansanten", 
    "resolution": "fixed", 
    "_ts": "1439238677802127", 
    "component": "combo core", 
    "summary": "[sim-services] I3MCTree throws when printing after removing nodes", 
    "priority": "major", 
    "keywords": "", 
    "time": "2015-07-08T23:22:29", 
    "milestone": "", 
    "owner": "olivas", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150708T23:32:29: olivas changed summary from "[sim-services] I3MCTree throws when printing after removing nodes from an I3MCTree" to "[sim-services] I3MCTree throws when printing after removing nodes"

**Comment by icecube-trac** [comment]
Trac update at 20150709T10:14:42: 
* jvansanten changed status from "new" to "closed"
* jvansanten commented:

> In changeset 134927:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="134927"
> Ensure that the head_ of I3MCTree is reset when the corresponding node is erased. Fixes #1038.
> ```


* jvansanten changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20150810T20:31:17: 
* olivas commented:

> Milestone offline V15-07-00 deleted


* olivas changed milestone from "offline V15-07-00" to ""
